src/osinfo.rs

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::collections::HashMap;
+
+use astarte_sdk::types::AstarteType;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+pub fn extract_field(re: &Regex, text: &str) -> String {
+    if let Some(cap) = re.captures(text) {
+        if let Some(mat) = cap.name("field") {
+            return mat.as_str().to_string();
+        }
+    }
+    String::new()
+}
+
+lazy_static! {
+    static ref OS_NAME_RE: Regex = Regex::new(r#"(?m)^NAME="(?P<field>.*)$"#).unwrap();
+    static ref OS_VERSION_RE: Regex =
+        Regex::new(r#"(?m)^(VERSION_ID=)|(BUILD_ID=)(?P<field>.*)$"#).unwrap();
+}
+
+/// io.edgehog.devicemanager.OSInfo
+pub fn get_os_info() -> Result<HashMap<String, AstarteType>, std::io::Error> {
+    let os = std::fs::read_to_string("/etc/os-release")?;
+
+    let mut ret: HashMap<String, AstarteType> = HashMap::new();
+
+    ret.insert(
```

Comment:
Sending an empty string as osName doesn't make sense, let's rather avoid sending it.